File: server/internal/model/record.go
package model

type Record struct {
	Timestamp  string  `json:"timestamp"`
	Percentage int16   `json:"percentage"`
	Video      *string `json:"video"`
}

type RecordWithUser struct {
	Record
	User User `json:"user"`
}

type RecordWithLevel struct {
	Record
	Level Level `json:"level"`
}

File: server/src/schema/record.ts
import { z } from 'zod';
import { User } from './user';
import { Level } from './level';

export const Record = z.object({
	timestamp: z.string().datetime(),
	percentage: z.number().int().min(0).max(100),
	video: z.string(),
});

export const RecordWithUser = Record.extend({
	user: User,
});

export const RecordWithLevel = Record.extend({
	level: Level,
});
